src/main/java/com/synopsys/integration/alert/provider/polaris/PolarisApiHelper.java

```diff
@@ -124,22 +126,26 @@ public PolarisApiHelper(final IssueService issueService, final RoleAssignmentsSe
         return Optional.empty();
     }
 
-    private Map<String, Integer> mapIssueTypeToCount(final List<QueryIssueResource> queryIssues) {
+    private Map<String, Integer> mapIssueTypeToCount(final String projectId, final String branchId, final List<IssueV0Resource> queryIssues) throws IntegrationException {
         final Map<String, Integer> issueTypeCounts = new HashMap<>();
-        for (final QueryIssueResource queryIssue : queryIssues) {
-            // FIXME issue type is not the same as issue key
-            final String issueType = queryIssue.getAttributes().getSubTool();
-            if (!issueTypeCounts.containsKey(issueType)) {
-                issueTypeCounts.put(issueType, 0);
-            }
-            final Integer tempCount = issueTypeCounts.get(issueType);
-            issueTypeCounts.put(issueType, tempCount + 1);
+        for (final IssueV0Resource queryIssue : queryIssues) {
+            final String issueKey = queryIssue.getAttributes().getIssueKey();
+            final IssueResourcesSingle populatedIssueResouce = issueService.getIssueForProjectBranchAndIssueKeyWithDefaultIncluded(projectId, branchId, issueKey);
+            final Optional<IssueTypeV0Resource> optionalIssueTypeResource = issueService.getIssueTypeFromPopulatedIssueResources(populatedIssueResouce);
+            optionalIssueTypeResource.ifPresent(issueTypeResource -> {
+                final String issueType = issueTypeResource.getAttributes().getName();
+                if (!issueTypeCounts.containsKey(issueType)) {
+                    issueTypeCounts.put(issueType, 0);
+                }
+                final Integer tempCount = issueTypeCounts.get(issueType);
+                issueTypeCounts.put(issueType, tempCount + 1);
+            });
```

Comment:
When do we want to use ifPresent vs an if statement? ifPresent is nice if we have to reference the object multiple times. But here we just reference the object once and only need to call a single get. So... what do we prefer?